thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/common/thrift/ThriftMetadataAccess.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common.thrift;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.thrift.TBase;
+import org.apache.thrift.TFieldIdEnum;
+import org.apache.thrift.meta_data.FieldMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+public final class ThriftMetadataAccess {
+
+    private static final Logger logger = LoggerFactory.getLogger(ThriftMetadataAccess.class);
+
+    private static boolean preInitializeThriftClass;
+    private static final Pattern preInitializeTargetPattern =
+            Pattern.compile("^armeria-thrift0\\.(\\d+)\\..*$");
+
+    static {
+        try {
+            final Enumeration<URL> versionPropertiesUrls =
+                    ThriftMetadataAccess.class.getClassLoader().getResources(
+                            "META-INF/com.linecorp.armeria.versions.properties");
+            boolean preInitializeThriftClass = false;
+            while (versionPropertiesUrls.hasMoreElements()) {
+                final URL url = versionPropertiesUrls.nextElement();
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+                    final Properties props = new Properties();
+                    props.load(reader);
+                    preInitializeThriftClass = needPreInitialize(props);
+                    if (preInitializeThriftClass) {
+                        break;
+                    }
+                }
+            }
+            ThriftMetadataAccess.preInitializeThriftClass = preInitializeThriftClass;
+        } catch (Exception e) {
+            logger.trace("Unexpected exception while determining the 'armeria-thrift' version: ", e);
+        }
+    }
+
+    @VisibleForTesting
+    static boolean needPreInitialize(Properties props) {
```

Comment:
Thrid-person verb is commonly used for boolean methods.
https://softwareengineering.stackexchange.com/questions/404650/what-form-of-verb-to-use-imperative-verb-or-third-person-verb-in-programming
```suggestion
    static boolean needsPreInitialization(Properties props) {
```